### PR TITLE
DEV-618 Add account factory architecture overview

### PR DESCRIPTION
**Pull request by Resonance1584**

### CODE REVIEW

**Comment by Resonance1584** [comment]
@yhakbar can I get a sanity check on these diagrams? I've intentionally tried to leave _out_ as much as possible so that they're easier to digest

**Comment by ZachGoldberg** [comment]
Any chance you have a source for the diagrams you can check in as well to make future modifications feasible?

File: docs/2.0/docs/accountfactory/architecture/index.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+   Pipelines ->> Core Accounts: Execute terragrunt to baseline account
+```
+
+## IAM Roles
+
+Each new account has a set of IAM policies that determine the scope of changes Pipelines can plan/apply within AWS. Pipelines will automatically assume the appropriate roles for each account when changes are detected. Read about the [roles in full here](/2.0/docs/pipelines/architecture/security-controls#roles-provisioned-by-devops-foundations).
+
+## Delegated Repositories
+
+Delegated repositories provide additional control over your infrastructure by expanding on the above architecture. When vending delegated repositories new account security baselines are still tracked in your `infrastructure-live-root` repository, however other infrastructure is tracked in a new repository specific to this account(s). New IAM roles are added to your `infrastructure-live-access-control` repository that control the allowed infrastructure changes that can take place via Pipelines in the delegated repository.
```
Comment (ZachGoldberg) [inline docs/2.0/docs/accountfactory/architecture/index.md:38]:
```suggestion
Delegated repositories provide additional control over your infrastructure by expanding on the above architecture. When vending delegated repositories new account security baselines are still tracked in your `infrastructure-live-root` repository, however other infrastructure is tracked in a new repository specific to this account(s). New IAM roles are added to your `infrastructure-live-access-control` repository that are inherited by pipelines when deploying infrastructure in the delegated repositories, allowing the central platform team to control what changes can be implemented via Pipelines in the delegated repository.
```

**Comment by yhakbar** [comment]
> @yhakbar can I get a sanity check on these diagrams? I've intentionally tried to leave _out_ as much as possible so that they're easier to digest

These diagrams look good to me. They communicate everything simply and effectively. The one thing I would recommend changing is to show the same stacked multiple thingy that you did for IAM Role Modules for everything that's in a one/many to many relationship.

These are the things that might be multiplied:

- New Account Directory
- Request File
- Base IAM Role Modules
- Delegated IAM Role Modules
- New Account
- (D)elegated Repository <-- you might want to call it `infrastructure-live-<delegated>` for consistency.

**Review by ZachGoldberg — APPROVED** [review]

**Review by ZachGoldberg — APPROVED** [review]

**Pull request merged**